Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift

```diff
@@ -72,239 +73,204 @@ public final class SwiftPackageManagerGraphGenerator: SwiftPackageManagerGraphGe
-                var cxxHeaderSearchPaths: [String] = []
-                var cDefines: [String: String] = [:]
-                var cxxDefines: [String: String] = [:]
-                var swiftDefines: [String: String] = [:]
-                var cFlags: [String] = []
-                var cxxFlags: [String] = []
-                var swiftFlags: [String] = []
-
-                try target.settings.forEach { setting in
-                    let platforms = try setting.condition?.platforms()
-                    switch (setting.tool, setting.name) {
-                    case (.c, .headerSearchPath):
-                        cHeaderSearchPaths.append(setting.value[0])
-                    case (.c, .define):
-                        let (name, value) = setting.extractDefine
-                        cDefines[name] = value
-                    case (.c, .unsafeFlags):
-                        cFlags.append(contentsOf: setting.value)
-
-                    case (.cxx, .define):
-                        let (name, value) = setting.extractDefine
-                        cxxDefines[name] = value
-                    case (.cxx, .headerSearchPath):
-                        cxxHeaderSearchPaths.append(setting.value[0])
-                    case (.cxx, .unsafeFlags):
-                        cxxFlags.append(contentsOf: setting.value)
-
-                    case (.swift, .define):
-                        let (name, value) = setting.extractDefine
-                        swiftDefines[name] = value
-                    case (.swift, .unsafeFlags):
-                        swiftFlags.append(contentsOf: setting.value)
-
-                    case (.linker, .linkedFramework):
-                        dependencies.append(.linkedFramework(name: setting.value[0], platforms: platforms))
-                    case (.linker, .linkedLibrary):
-                        dependencies.append(.linkedLibrary(name: setting.value[0], platforms: platforms))
-                    case (.c, .linkedFramework), (.c, .linkedLibrary), (.cxx, .linkedFramework), (.cxx, .linkedLibrary),
-                         (.swift, .headerSearchPath), (.swift, .linkedFramework), (.swift, .linkedLibrary),
-                         (.linker, .headerSearchPath), (.linker, .define), (.linker, .unsafeFlags):
-                        throw SwiftPackageManagerGraphGeneratorError.unsupportedSetting(setting.tool, setting.name)
+extension PackageInfo.Target {
+    func mapPlatform() -> ProjectDescription.Platform {
+        // TODO: Should this be configured in `Dependencies.swift`?
+        return .iOS
+    }
+
+    func mapProduct(packageInfo: PackageInfo) -> ProjectDescription.Product? {
+        return packageInfo.products
+            .filter { $0.targets.contains(self.name) }
+            .compactMap {
+                switch $0.type {
+                case let .library(type):
+                    switch type {
+                    case .automatic:
+                        return .staticLibrary
```

Comment:
we should either:
a) try to do this automatically as SPM does
b) let the users specify whether they want the SPM dependency linked dynamically or statically - defined in `Dependencies.swift`

I am probably more inclined to option b) as it gives more control to the user, however, we will depart from how SPM works.

Sorry if I misunderstood what `.automatic` stands for here.

Reply:
Added a warning to handle it in a separate PR

Reply:
Addressed here: https://github.com/tuist/tuist/pull/3105